crates/wasi-common/src/error.rs

```diff
@@ -242,6 +248,280 @@ impl Error {
     pub const ENOTCAPABLE: Self = Error::Wasi(WasiError::ENOTCAPABLE);
 }
 
+impl From<Error> for std::io::Error {
+    fn from(err: Error) -> std::io::Error {
+        match err {
+            Error::Wasi(e) => e.into(),
+            Error::Io(e) => e,
+            #[cfg(unix)]
+            Error::Yanix(e) => std::io::Error::new(std::io::ErrorKind::Other, e),
+        }
+    }
+}
+
 pub(crate) trait FromRawOsError {
     fn from_raw_os_error(code: i32) -> Self;
 }
+
+/// Translate a WASI errno code into an `io::Result<()>`.
+///
+/// TODO: Would it be better to have our own version of `io::Error` (and
+/// `io::Result`), rather than trying to shoehorn WASI errors into the
+/// libstd version?
+fn wasi_errno_to_io_error(errno: wasi::__wasi_errno_t) -> std::io::Error {
```

Comment:
This looks the wrong way round for me. As a temp solution, this is fine, but in the long run either, as you suggested, we should have our own version of `io::Error` or flatten all errors into WASI errno.

Actually, why do we need to convert the errno into `io::Error` in the first place? As far as I can remember, we've only ever done it the opposite way in this crate.

Reply:
OK, I've only now realised this is a simple copy-paste from `fs/error.rs`. Nvm then :-)

Reply:
It's true yeah it was a copy/paste, but I went ahead and refactored a bit more and was able to delete this function entirely.